client/src/components/signing-form/signing-form.js

```diff
@@ -0,0 +1,47 @@
+import ko from 'knockout';
+
+import 'components/signing-form/signing-form.less';
+import template from 'components/signing-form/signing-form.html';
+import register from 'components/component-registrator';
+
+import {user} from 'providers/user-provider';
+
+class UserActionsViewModel {
+    constructor(params) {
+        this.mode = params.mode;
+        this.user = params.user;
+        this.mode('sign-in');
+        this.password = ko.observable('');
+        this.email = ko.observable('');
+        this.name = ko.observable('');
+    }
+
+    isSignUp() {
+        return this.mode() === 'sign-up';
+    }
+
+    toggleMode() {
+        this.mode(this.mode() === 'sign-up' ? 'sign-in' : 'sign-up');
```

Comment:
These string literals appear everywhere. It may be worth it to move them to constants.